lib/rules/no-whitespace-before-property.js

```diff
@@ -25,14 +46,37 @@ module.exports = function(context) {
             var prop = node.property;
 
             if (astUtils.isTokenOnSameLine(obj, prop)) {
-                if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
-                    context.report({
-                        node: node,
-                        message: "Unexpected whitespace before property '{{ propName }}'.",
-                        data: {
-                            propName: prop.name
+                var tokenBeforeProp;
+                var tokenBeforePunc;
+
+                tokenBeforeProp = sourceCode.getTokenBefore(prop);
+                while (tokenBeforeProp.value !== "." && tokenBeforeProp.value !== "[") {
+                    tokenBeforeProp = sourceCode.getTokenBefore(tokenBeforeProp);
+                }
+
+                tokenBeforePunc = sourceCode.getTokenBefore(tokenBeforeProp);
+
+                if (tokenBeforeProp.value === ".") {
```

Comment:
Hey @mysticatea, thanks for the feedback :) `leftToken` and `rightToken` is much clearer - I like that! Maybe I'm missing something, but each conditional branch checks for whitespace between different tokens - no two are the same. I'd love to simplify it down to what you have in your example above, but I'm not seeing a way to do that. Mind explaining?

I'm going to change variable names right now though!